x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java

```diff
@@ -565,7 +566,7 @@ static DeprecationIssue emptyDataTierPreferenceCheck(ClusterState clusterState,
                         + DataTier.TIER_PREFERENCE
                         + "] setting, "
                         + "in 8.0 this setting will be required for all indices and may not be empty or null.",
-                    "https://www.elastic.co/guide/en/elasticsearch/reference/current/data-tiers.html",
+                    "https://ela.st/es-deprecation-7-empty-tier-preference",
```

Comment:
I'd init cap index in the deprecation message, and tweak the description slightly:

Index "indexName" does not have a "DataTier.TIER_PREFERENCE" configured

A data tier preference must be specified for all indices in 8.0. Set a tier preference for "indexName".

Reply:
Good catch. Fixed.